codebase/src/compiler/checker-relations.md

```diff
@@ -0,0 +1,90 @@
+# Relations
+
+There are 5 different type relations we have in the checker (in order of strictness, i.e. each is a subset of the next):
+- Identity
+- Strict subtyping
+- Subtyping
+- Assignability
+- Comparability
+
+We use those relations for different things in the checker, and in general we have all those different relations because sometimes we wanna be more strict, and sometimes we don't.
+
+We use **assignability** to check if an argument can be passed to a function, or if an expression can be assigned to a variable:
```

Comment:
This is true - but we use assignability for almost everything. Even for overriding types.

```ts
interface Base<T> {
    value: { prop?: T };
}

interface Derived<T> extends Base<T> {
    value: {};
}
```